ModManager/VersionSystem/ModIoFileExtensions.cs

```diff
@@ -1,15 +1,33 @@
-﻿using System.Text.RegularExpressions;
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
 using Modio.Models;
 
 namespace ModManager.VersionSystem
 {
     public static class ModIoFileExtensions
     {
+        private static readonly string CompatibleVersionArrayPattern = @"CompatibleGameVersions:\s*\[\s*([\s\S]*?)\s*\]\s*";
+        private static readonly Regex CompatibleVersionArrayRegex = new(CompatibleVersionArrayPattern, RegexOptions.Singleline);
+        
+        private static readonly string CompatibleVersionPattern = "\"(\\d+(\\.\\d+)*)+\"";
+        private static readonly Regex CompatibleVersionRegex = new(CompatibleVersionPattern);
+        
+        public static IEnumerable<string?> CompatibleGameVersions(this File file)
+        {
+            if (string.IsNullOrEmpty(file.Changelog) || string.IsNullOrEmpty(file.MetadataBlob))
```

Comment:
Should this be checking for Changelog? It's not otherwise referenced in this function.